skrf/network.py

```diff
@@ -1305,30 +1306,95 @@ def group_delay(self):
 
         return gd
 
-    ## NETWORK CLASIFIERs
-    def is_reciprocal(self):
+    ## NETWORK CLASSIFIERs
+    def is_reciprocal(self, tol=mf.ALMOST_ZERO):
         '''
         test for reciprocity
         '''
-        raise (NotImplementedError)
+        return npy.allclose(reciprocity(self.s), npy.zeros_like(self.s), atol=tol)
 
-    def is_symmetric(self):
-        '''
-        test for symmetry
+    def is_symmetric(self, n=1, port_order={}, tol=mf.ALMOST_ZERO):
```

Comment:
i think any method of expressing  symmetry will be complicated due to the number of possibilities, and confusions with port indecies and numbering. if this feature becomes popular we may consider writing an example notebook in the  docs to demonstrate it.  i think you tests below are good at this as well.